selfdrive/ui/qt/api.cc

```diff
@@ -72,76 +72,67 @@ QString CommaApi::create_jwt(QVector<QPair<QString, QJsonValue>> payloads, int e
   return jwt;
 }
 
-RequestRepeater::RequestRepeater(QWidget* parent, QString requestURL, int period_seconds, const QString &cache_key, bool disableWithScreen)
-  : disableWithScreen(disableWithScreen), cache_key(cache_key), QObject(parent)  {
-  networkAccessManager = new QNetworkAccessManager(this);
+std::pair<QNetworkReply::NetworkError, QString> httpGet(const QString& url, int timeout_ms, QMap<QString, QString>* headers) {
+  QNetworkRequest request(QUrl{url});
+  if (headers) {
+    for (auto it = headers->constBegin(); it != headers->constEnd(); ++it) {
+      request.setRawHeader(it.key().toUtf8(), it.value().toUtf8());
+    }
+  }
+#ifdef QCOM
+  QSslConfiguration ssl = QSslConfiguration::defaultConfiguration();
+  ssl.setCaCertificates(QSslCertificate::fromPath("/usr/etc/tls/cert.pem", QSsl::Pem, QRegExp::Wildcard));
+  request.setSslConfiguration(ssl);
+#endif
+
+  QEventLoop loop;
```

Comment:
is it safe to create another event loop outside the main one? doesn't this make the requests synchronous now?

Reply:
it's safe,the function is "block", but all events will be dispatched to their usual places during waiting.